client/dashboard/plugins/plugin/index.tsx

```diff
@@ -0,0 +1,263 @@
+					},
+				},
+			] }
+			getItemId={ ( item ) => String( item.ID ) }
+			paginationInfo={ paginationInfo }
+		/>
+	);
+};
+
+const defaultSitesWithoutThisPluginView: View = {
+	type: 'table',
+	fields: [ 'link' ],
+	layout: {
+		styles: {
+			link: { align: 'end' },
+		},
+	},
+	sort: { field: 'name', direction: 'asc' },
+	titleField: 'domain',
+};
+
+const SitesWithoutThisPlugin = ( {
+	sitesWithoutThisPlugin,
+}: {
+	sitesWithoutThisPlugin: Site[];
+} ) => {
+	const [ view, setView ] = useState< View >( defaultSitesWithoutThisPluginView );
+	const { pluginId } = pluginRoute.useParams();
+	const locale = useLocale();
+	const { isLoading: isLoadingSitesPlugins } = useQuery( pluginsQuery() );
+	const { isLoading: isLoadingSites } = useQuery( sitesQuery() );
+	const { isLoading: isLoadingWpOrgPlugin } = useQuery( wpOrgPluginQuery( pluginId, locale ) );
+
+	const fields = useMemo(
+		() => [
+			{
+				id: 'domain',
+				label: __( 'Site' ),
+				getValue: ( { item }: { item: Site } ) => item.URL,
+				render: ( { item }: { item: Site } ) => item.URL,
+				enableHiding: false,
+				enableSorting: true,
+				enableGlobalSearch: true,
+			},
+			{
+				id: 'link',
+				header: <div />,
+				getValue: ( { item }: { item: Site } ) => item.URL,
+				render: ( { item }: { item: Site } ) => (
+					<Link to={ `/plugins/${ pluginId }/${ item.slug }` }>{ __( 'Go to plugin page' ) }</Link>
+				),
+			},
+		],
+		[ pluginId ]
+	);
+
+	const { data, paginationInfo } = filterSortAndPaginate( sitesWithoutThisPlugin, view, fields );
+
+	return (
+		<DataViews
+			search={ false }
+			isLoading={ isLoadingSitesPlugins || isLoadingSites || isLoadingWpOrgPlugin }
+			data={ data }
+			fields={ fields }
+			view={ view }
+			onChangeView={ setView }
+			defaultLayouts={ { table: {} } }
+			getItemId={ ( item ) => String( item.ID ) }
+			paginationInfo={ paginationInfo }
+		>
+			<DataViews.Layout />
+			<DataViews.Pagination />
+		</DataViews>
+	);
+};
+
+export default function Plugin() {
+	const { pluginId } = pluginRoute.useParams();
+	const locale = useLocale();
+	const { data: sitesPlugins, isLoading: isLoadingSitesPlugins } = useQuery( pluginsQuery() );
+	const { data: sites, isLoading: isLoadingSites } = useQuery( sitesQuery() );
+	const { data: wpOrgPlugin, isLoading: isLoadingWpOrgPlugin } = useQuery(
+		wpOrgPluginQuery( pluginId, locale )
+	);
+
+	const siteIdsWithThisPlugin = sitesPlugins?.sites
+		? Object.entries( sitesPlugins.sites ).flatMap( ( [ siteId, plugins ] ) =>
+				plugins.some( ( p ) => p.slug === pluginId ) ? [ siteId ] : []
+		  )
+		: [];
+
+	const [ sitesWithThisPlugin, sitesWithoutThisPlugin ] = sites
+		? sites.reduce(
+				( acc, site ) => {
+					if ( siteIdsWithThisPlugin.includes( String( site.ID ) ) ) {
+						acc[ 0 ].push( site );
+					} else {
+						acc[ 1 ].push( site );
+					}
+					return acc;
+				},
+				[ [], [] ] as [ Site[], Site[] ]
+		  )
+		: [ [], [] ];
```

Comment:
We compute the data here, and then each component re-computes the same (or very similar) data?

Reply:
Nope, each component is fed one part of the partition.